### DISCUSSION TITLE
Webstorm complains Module is not listed in package.json dependencies

### DISCUSSION THREAD

**Discussion by spinlokd** [quasarframework/quasar#10976] [General - Components / Directives / etc]
Used quasar CLI to create a new app, and it created package.json with the following dependencies:
```json
  "dependencies": {
    "@quasar/extras": "^1.0.0",
    "axios": "^0.21.1",
    "core-js": "^3.6.5",
    "quasar": "^2.0.0",
    "vuex": "^4.0.1"
  },
```
When I open .vue files, I get "Module is not listed in package.json dependencies" for 
```js
import { defineComponent } from 'vue';
```
When I open index.js, I get same warning for 
```js
import { createRouter, createMemoryHistory, createWebHistory, createWebHashHistory } from 'vue-router'
```
Is this because WebStorm doesn't understand dependencies that come with quasar? Should I explicitly specify (dev)dependencies on vue, vue-router, etc?

**Comment by hawkeye64** [comment]
@spinlokd In your `devDependencies` you will see a `@quasar/app`. This is the package that contains all of the pinned versions of other packages that are required for your Quasar app to build and run without issues.
Both Yarn and Npm hoist packages to your `node_modules` folder, but also be aware that they have slightly different algorithms for this, so running Yarn or Npm (one or the other) may produce different results.
This really isn't a Quasar issue, but WebStorm. Also, they may have a proprietary format for this as well (as the VS Code plugin `Vetur` does).
